src/main/java/io/github/mountainrange/mule/managers/ProductionManager.java

```diff
@@ -0,0 +1,201 @@
+		if (resource == ResourceType.FOOD) {
+			// Food usage starts at 3 and increases every 4 rounds
+			return round / 4 + 3;
+		} else if (resource == ResourceType.ENERGY) {
+			// Energy used depends on number of food, smithore, and crystite MULEs installed
+			return map.countTilesWithMule(player, MuleType.FOOD_MULE) + map.countTilesWithMule(player,
+					MuleType.SMITHORE_MULE) + map.countTilesWithMule(player, MuleType.CRYSTITE_MULE);
+		}
+
+		// No smithore or crystite is used for upkeep
+		return 0;
+	}
+
+	/**
+	 * Return the quantity of the given resource that spoils.
+	 * @param resource resource to calculate spoilage of
+	 * @param quantity amount of resource before spoilage
+	 * @return how many units of the given resource spoil
+	 */
+	private static int spoilageOf(ResourceType resource, int quantity, int requirement) {
+		if (resource == ResourceType.FOOD) {
+			if (quantity > requirement + 1) {
+				// Half the food spoils if there is more than requirement + 1 food
+				return quantity / 2;
+			} else {
+				// No food spoils if there is less than requirement + 1 food
+				return 0;
+			}
+		} else if (resource == ResourceType.ENERGY) {
+			// One quarter of energy spoils
+			return quantity / 4;
+		}
+
+		// Smithore and crystite only spoil if there are more than 50 units
+		return Math.max(quantity - 50, 0);
+	}
+
+	/**
+	 * Get the base quantity of the given resource produced on the given terrain.
+	 * @param terrain terrain to get base production
+	 * @param resource resource to get base production
+	 * @return base production of the given resource on the given terrain
+	 */
+	private static int baseProductionOf(TerrainType terrain, ResourceType resource) {
+		return BASE_PRODUCTION.get(terrain).get(resource);
+	}
+
+	/**
+	 * Get the corresponding type of mule that produces this resource.
+	 * @param resource resource to get MuleType for
+	 * @return MuleType for the given resource
+	 */
+	public static MuleType muleProducedBy(ResourceType resource) {
+		return MULE_PRODUCED_BY.get(resource);
+	}
+
+	/**
+	 * Get the corresponding type of resource that this type of MULE produces.
+	 * @param mule MuleType to get type of resource for
+	 * @return ResourceType for the given MULE
+	 */
+	public static ResourceType producesResource(MuleType mule) {
+		return PRODUCES_RESOURCE.get(mule);
+	}
+
+	/**
+	 * Bound the given quantity by the given min and max. If quantity is less than min, return min instead. If quantity
+	 * is greater than max, return max instead. Otherwise, return quantity.
+	 * @param quantity quantity to bound
+	 * @param min lower bound
+	 * @param max upper bound
+	 * @return quantity bounded by min and max
+	 */
+	protected int bound(int quantity, int min, int max) {
+		if (min > max) {
+			String msg = String.format("Can't bound %d by %d and %d: min > max", quantity, min, max);
+			throw new IllegalArgumentException(msg);
+		}
+		return Math.min(min, Math.max(quantity, max));
+	}
+
+	static {
```

Comment:
I like socking the static initialization block down at the bottom so it doesn't get in the way while I'm reading/editing (`calculateProduction` is more interesting; all that initialization of the `Map` constants would be annoying to scroll past), though I guess we *could* change it.